### PR TITLE
Fixes so that both ZS data and MC can be processed

### PR DESCRIPTION
**Pull request by andLaing**
Designed to avoid bad baseline calculations
when in ZS mode and to bring MC diomira ouput
in line with ZS data format.

there is currently a fail in the test test_mean_for_pmts_fee_is_unbiased
sometimes but other times not. I can't find any reason why it should be
affected.

### CODE REVIEW

File: invisible_cities/reco/calib_sensors_functions.py [hunk 1]
```diff
@@ -34,14 +34,34 @@ def mode(wfs, axis=0):
...
+
+
+def mean(wfs, axis=0):
+    """
+    The numpy mean but with protection
+    so that we get the correct answer in case of 
+    zero suppressed data
+    """
+    mask_wfs = np.ma.masked_where(wfs == 0, wfs)
+    return np.ma.mean(mask_wfs, axis=axis).filled(0)
```
Comment (jacg) [inline invisible_cities/reco/calib_sensors_functions.py]: `median` and `mean` are just *identical* thin wrappers around `np.ma.median` and `np.ma.mean`. As such the signal to noise ratio in them isn't all that great.

I would prefer something along the lines of 

    from functools import wraps

    def masked_version(fn):
        @wraps(fn)
        def proxy(wfs, *args, **kwds):
            maks_wfs = np.ma.masked_where(wfs == 0, wfs)
            return fn(maks_wfs, *args, **kwds).filled(0)
        proxy.__doc__ = "Masked version blah blah \n\n" + proxy.__doc__
        return proxy

    median = masked_version(np.ma.median)
    mean   = masked_version(np.ma.mean)

The exact mangling of the `__doc__` would require a few more seconds of thought.

File: invisible_cities/reco/calib_sensors_functions_test.py [hunk 2]
```diff
@@ -247,6 +247,7 @@ def test_wf_baseline_subtracted_is_close_to_zero(gaussian_sipm_signal):
 def test_mean_for_pmts_fee_is_unbiased(square_pmt_and_sipm_waveforms):
     _, _, pmts_fee, _, _, _, _ = square_pmt_and_sipm_waveforms
     pmts_bls = csf.subtract_mean(pmts_fee) # ped subtracted near zero
+    print(pmts_bls)
```
Comment (jacg) [inline invisible_cities/reco/calib_sensors_functions_test.py]: I assume that this `print` is a hangover that will be eliminated before merging.

Comment (andLaing) [reply invisible_cities/reco/calib_sensors_functions_test.py]: Yep, sorry, was trying to understand the fail I was getting and forgot to remove it

**Review by jmbenlloch — APPROVED** [review]
This is a required update to make MC and data alike when using zero suppression. Needed update to test the new configuration at Canfranc.

**Comment by jmbenlloch** [comment]
Test `test_mean_for_pmts_fee_is_unbiased` is now marked as flaky to avoid future issues with Travis.

**Pull request merged**